### PR TITLE
Translated Context.md

### PR DESCRIPTION
**Pull request by tamarau**
Thanks for reviewing! 🙂

### CODE REVIEW

**Comment by ph1p** [comment]
I think we should write "Context" instead of "Kontext", because this is a technical term in React and all methods are named after it. (setContext etc.)

**Comment by tamarau** [comment]
Hi Phil, yeah I spent quite a bit thinking about whether to use Kontext or Context. Switched it back to Context 👍

File: examples/context/updating-nested-context-context.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-// Make sure the shape of the default value passed to
-// createContext matches the shape that the consumers expect!
+// Stelle sicher, dass die Form des Default-Wertes, welcher an createContext
+// überreicht wird mit der Form die der Konsument erwarter, übereinstimmt!
```
Comment (ph1p) [inline examples/context/updating-nested-context-context.js]:
```suggestion
// überreicht wird mit der Form die der Konsument erwartet, übereinstimmt!
```

File: content/docs/context.md [hunk 2]
```diff
@@ -4,58 +4,59 @@ title: Context
 permalink: docs/context.html
 ---
 
-Context provides a way to pass data through the component tree without having to pass props down manually at every level.
+Context ermöglicht es, Daten durch den Komponentenbaum zu übergeben, ohne die Props in jeder Schicht manuell zu übergeben.
```
Comment (ph1p) [inline content/docs/context.md]:
```suggestion
Context ermöglicht es, Daten durch den Komponentenbaum zu leiten, ohne die Props in jeder Schicht manuell zu übergeben.
```

File: content/docs/context.md [hunk 3]
```diff
@@ -4,58 +4,59 @@ title: Context
...
 
-In a typical React application, data is passed top-down (parent to child) via props, but this can be cumbersome for certain types of props (e.g. locale preference, UI theme) that are required by many components within an application. Context provides a way to share values like these between components without having to explicitly pass a prop through every level of the tree.
+In einer typischen React-Applikation werden Daten von oben herab (Elternkomponete zu Kindkomponente) durch Props übergeben. Für bestimmte Arten von Props (z.B. lokale Preferenzen, UI Theme), welche von vielen Komponenten innerhalb der Applikation benötigt werden kann dies sehr umständlich werden. Mit Context ist es möglich, solche Werte zwischen Komponenten zu teilen, ohne diese explizit als Prop durch alle Schichten des Baumes zu übergeben.
```
Comment (ph1p) [inline content/docs/context.md]:
```suggestion
In einer typischen React-Anwendung werden Daten top-down (Elternkomponente zurKindkomponente) durch Props übergeben. Für bestimmte Arten von Props (z.B. lokale Einstellungen, UI Theme), welche von vielen Komponenten innerhalb der Anwendung benötigt werden kann dies sehr umständlich werden. Mit Hilfe von Context ist es möglich, solche Werte zwischen Komponenten zu teilen, ohne diese explizit als Prop durch alle Schichten des Baumes zu geben. 
```

File: content/docs/context.md [hunk 4]
```diff
@@ -4,58 +4,59 @@ title: Context
...
-  - [Context.displayName](#contextdisplayname)
-- [Examples](#examples)
-  - [Dynamic Context](#dynamic-context)
-  - [Updating Context from a Nested Component](#updating-context-from-a-nested-component)
-  - [Consuming Multiple Contexts](#consuming-multiple-contexts)
-- [Caveats](#caveats)
+- [Beispiele](#examples)
+  - [Dynamischer Context](#dynamic-context)
+  - [Updaten des Contextes von einer genesteten Komponente](#updating-context-from-a-nested-component)
+  - [Konsumieren mehrerer Contexten](#consuming-multiple-contexts)
```
Comment (ph1p) [inline content/docs/context.md]:
```suggestion
  - [Konsumieren mehrerer Contexte](#consuming-multiple-contexts)
```

File: content/docs/context.md [hunk 5]
```diff
@@ -4,58 +4,59 @@ title: Context
...
+- [Fallen](#caveats)
 - [Legacy API](#legacy-api)
 
-## When to Use Context {#when-to-use-context}
+## Wann Context verwendet werden soll {#when-to-use-context}
 
-Context is designed to share data that can be considered "global" for a tree of React components, such as the current authenticated user, theme, or preferred language. For example, in the code below we manually thread through a "theme" prop in order to style the Button component:
+Context wurde entwickelt, um Daten zu teilen, welche für einen Baum von React Komponenten als "global" bezeichnet werden können, zum Beispiel der aktelle authentifizierter Nutzer, das Theme oder die bevorzugte Sprache. In dem unten angeführten Codebeispiel zum Beispiel, ziehen wir ein Theme Prop manuell durch den Code, um die Button-Komponente zu stylen:
```
Comment (ph1p) [inline content/docs/context.md]:
```suggestion
Context wurde entwickelt, um Daten zu teilen, welche für einen Baum von React Komponenten als "global" bezeichnet werden können, zum Beispiel der aktuelle authentifizierte Nutzer, das Theme oder die bevorzugte Sprache. In dem unten angeführten Codebeispiel, leiten wir ein Theme Prop manuell durch den Code, um die Button-Komponente zu stylen:
```

File: content/docs/context.md [hunk 6]
```diff
@@ -4,58 +4,59 @@ title: Context
...
-Using context, we can avoid passing props through intermediate elements:
+Mit der Verwendung von Context können wir das Übergeben der Props durch Zwischenelemente verhindern:
 
 `embed:context/motivation-solution.js`
 
-## Before You Use Context {#before-you-use-context}
+## Bevor du Context verwendest {#before-you-use-context}
 
-Context is primarily used when some data needs to be accessible by *many* components at different nesting levels. Apply it sparingly because it makes component reuse more difficult.
+Context wird hauptsächlich verwendet, wenn Daten von *vielen* Komponenten in verschiedenen genesteten Schichten erreichbar sein müssen. Benütze es sparsam, weil es das Wiederverwenden von Komponenten schwieriger macht.
```
Comment (ph1p) [inline content/docs/context.md]:
```suggestion
Context wird hauptsächlich verwendet, wenn Daten von *vielen* Komponenten in verschiedenen genesteten Schichten erreichbar sein müssen. Benutze es sparsam, weil es das Wiederverwenden von Komponenten schwieriger macht.
```

File: content/docs/context.md [hunk 7]
```diff
@@ -68,21 +69,22 @@ function Page(props) {
...
+Mit dieser Veränderung muss nur die obereste Seitenkomponente über die Verwendung der `user` und `avatarSize` Props der `Link` und `Avatar` Komponenten Bescheid wissen.
 
-This *inversion of control* can make your code cleaner in many cases by reducing the amount of props you need to pass through your application and giving more control to the root components. However, this isn't the right choice in every case: moving more complexity higher in the tree makes those higher-level components more complicated and forces the lower-level components to be more flexible than you may want.
+Diese *inversion of control* (engl. für Umkehrung der Steuerung) macht den Code, durch das Reduzieren der Anzahl an überreichten Props  durch die Applikation und durch die Steigerung der Kontrolle der Wurzelkomponenten, in vielen Fällen sauberer. Jedoch ist das nicht in jedem Fall die richtige Entscheidung: durch das Schieben der Komplexität höher in den Baum, werden diese Higher-Level-Komponenten komplizierter und erzwingen die Lower-Level-Komponenten flexibler als erwünscht zu sein.
```
Comment (ph1p) [inline content/docs/context.md]:
```suggestion
Diese *inversion of control* (engl. für Umkehrung der Steuerung) macht den Code, durch das Reduzieren der Anzahl an überreichten Props durch die Anwendung und durch die Steigerung der Kontrolle der Root-Komponenten, in vielen Fällen sauberer. Jedoch ist das nicht in jedem Fall die richtige Entscheidung: durch das Schieben der Komplexität höher in den Baum, werden diese Higher-Level-Komponenten komplizierter und erzwingen die Lower-Level-Komponenten flexibler als erwünscht zu sein.
```

File: content/docs/context.md [hunk 8]
```diff
@@ -68,21 +69,22 @@ function Page(props) {
...
+
+Du bist nicht auf ein einzelnes Kind für eine Komponente beschränkt. Du kann mehrere Kinder durchgeben, sogar mehrere seperate Slots für Kinder aben, [wie hier dokumentiert wird](/docs/composition-vs-inheritance.html#containment):
```
Comment (ph1p) [inline content/docs/context.md]:
```suggestion
Du bist nicht auf ein einzelnes Kind für eine Komponente beschränkt. Du kannst mehrere Kinder durchgeben, sogar mehrere seperate Slots für Kinder haben, [wie hier dokumentiert wird](/docs/composition-vs-inheritance.html#containment):
```

**Comment by ph1p** [comment]
Hello @tamarau . Thank you for your translation (: Could you please read the pages again. Sometimes it sounds like a little bit to "Google-Translatorish" :/

**Comment by tamarau** [comment]
Hi @ph1p thanks for reviewing and will do! I know, after living in the UK for so long my German can sound a bit too Denglisch sometimes :(

File: examples/context/motivation-problem.js [hunk 9]
```diff
@@ -5,11 +5,11 @@ class App extends React.Component {
...
 function Toolbar(props) {
-  // highlight-range{1-4,7}
-  // The Toolbar component must take an extra "theme" prop
-  // and pass it to the ThemedButton. This can become painful
-  // if every single button in the app needs to know the theme
-  // because it would have to be passed through all components.
+  // highlight-range{1-4,7}  
+  // Diese Toolbar Komponente muss eine extra "theme" Prop annehmen
+  // und an ThemedButton übergeben. Das kann sehr umständlich werden,
+  // wenn jeder einzelne Button in der App über das Theme Bescheid
```
Comment (ph1p) [inline examples/context/motivation-problem.js]:
```suggestion
  // wenn jeder einzelne Button in der App über das Theme bescheid
```

File: examples/context/motivation-solution.js [hunk 10]
```diff
@@ -1,15 +1,15 @@
 // highlight-range{1-4}
-// Context lets us pass a value deep into the component tree
-// without explicitly threading it through every component.
-// Create a context for the current theme (with "light" as the default).
+// Context lasst uns einen Wert tief durch den Komponenten-Baum
+// übergeben ohne ihn explizit durch jede Komponente durchzureichen.
+// Erstelle einen Context für das aktuelle Theme (mit "light" als den Default).
```
Comment (ph1p) [inline examples/context/motivation-solution.js]:
```suggestion
// Erstelle einen Context für das aktuelle Theme (mit "light" als Standardwert).
```

File: examples/context/motivation-solution.js [hunk 11]
```diff
@@ -1,15 +1,15 @@
...
 const ThemeContext = React.createContext('light');
 
 class App extends React.Component {
   render() {
     // highlight-range{1-3,5}
-    // Use a Provider to pass the current theme to the tree below.
-    // Any component can read it, no matter how deep it is.
-    // In this example, we're passing "dark" as the current value.
+    // Verwende einen Provider um das aktuelle Theme durch den Baum weitergeben.
```
Comment (ph1p) [inline examples/context/motivation-solution.js]:
```suggestion
    // Verwende einen Provider um das aktuelle Theme durch den Baum zu leiten.
```

File: examples/context/motivation-solution.js [hunk 12]
```diff
@@ -1,15 +1,15 @@
...
+    // Jede Komponente kann es lesen, ganz egal wie tief sie sind.
```
Comment (ph1p) [inline examples/context/motivation-solution.js]:
```suggestion
    // Jede Komponente kann es lesen, ganz egal wie tief sie liegt.
```

File: examples/context/multiple-contexts.js [hunk 13]
```diff
@@ -1,7 +1,7 @@
-// Theme context, default to light theme
+// Theme Context, Defaultwert ist "light"
```
Comment (ph1p) [inline examples/context/multiple-contexts.js]:
```suggestion
// Theme Context, Standardwert ist "light"
```

File: examples/context/theme-detailed-theme-context.js [hunk 14]
```diff
@@ -11,5 +11,5 @@ export const themes = {
 
 // highlight-range{1-3}
 export const ThemeContext = React.createContext(
-  themes.dark // default value
+  themes.dark // default Wert
```
Comment (ph1p) [inline examples/context/theme-detailed-theme-context.js]:
```suggestion
  themes.dark // Standardwert
```

File: examples/context/updating-nested-context-context.js [hunk 15]
```diff
@@ -1,5 +1,5 @@
-// Make sure the shape of the default value passed to
-// createContext matches the shape that the consumers expect!
+// Stelle sicher, dass die Form des Default-Wertes, welcher an createContext
```
Comment (ph1p) [inline examples/context/updating-nested-context-context.js]:
```suggestion
// Stelle sicher, dass die Form des Standardwertes, welcher an createContext
```

File: examples/context/updating-nested-context-theme-toggler-button.js [hunk 16]
```diff
@@ -2,8 +2,8 @@ import {ThemeContext} from './theme-context';
 
 function ThemeTogglerButton() {
   // highlight-range{1-2,5}
-  // The Theme Toggler Button receives not only the theme
-  // but also a toggleTheme function from the context
+  // Der Theme Toggler Button erhält nicht nur das Theme, 
+  // sondern auch eine toggleTheme Funktion von Context
```
Comment (ph1p) [inline examples/context/updating-nested-context-theme-toggler-button.js]:
```suggestion
  // sondern auch eine toggleTheme-Funktion von Context
```

File: examples/context/motivation-problem.js [hunk 17]
```diff
@@ -5,11 +5,11 @@ class App extends React.Component {
 }
 
 function Toolbar(props) {
-  // highlight-range{1-4,7}
-  // The Toolbar component must take an extra "theme" prop
-  // and pass it to the ThemedButton. This can become painful
-  // if every single button in the app needs to know the theme
-  // because it would have to be passed through all components.
+  // highlight-range{1-4,7}  
+  // Diese Toolbar Komponente muss eine extra "theme" Prop annehmen
```
Comment (ph1p) [inline examples/context/motivation-problem.js]:
```suggestion
  // Diese Toolbar-Komponente muss eine extra "theme" Prop annehmen
```

**Comment by ph1p** [comment]
Thank you! Please check my suggestion and I'll merge it as soon as possible (:

**Comment by tamarau** [comment]
@ph1p all done! :)

**Pull request merged**